File: PTA/Advanced_Level/1036.cpp
#include<bits/stdc++.h>
using namespace std;

int main() {
	string name, gender, id;
	int score, m_score, f_score;
	string m, f, m_id, f_id;
	int t;
		cin >> t;
	while(t--) {
		cin >> name >> gender >> id >> score;
		if(gender == "M" && (!m.size() || m_score > score)) {
			m_score = score, m = name;
			m_id = id;
		}
		if(gender == "F" && (!f.size() || f_score < score)) {
			f_score = score, f = name;
			f_id = id;
		}
	}
	if(f.empty()) puts("Absent");
	else cout << f << ' ' << f_id << endl;
	if(m.empty()) puts("Absent");
	else cout << m << ' ' << m_id << endl;
	if(f.size() && m.size()) cout << f_score - m_score << endl;
	else puts("NA");
	return 0;
}

File: PTA/Advanced_Level/1036.rs
use std::io::stdin;

fn main() {
    
    let mut input: String = String::new();
    stdin().read_line(&mut input).unwrap();
    let mut t: u32 = input.trim().parse::<u32>().unwrap();
    input.clear();

    let mut m: String = String::new();
    let mut f: String = String::new();
    let mut m_id: String = String::new();
    let mut f_id: String = String::new();
    let mut m_score: i32 = 101;
    let mut f_score: i32 = 0;

    for _ in 0..t {
        stdin().read_line(&mut input).unwrap();
        let v: Vec<String> = input.trim().split(' ').map(|s| s.to_string()).collect();
        let score: i32 = v[3].parse::<i32>().unwrap();
        //println!("{}", score);
        if v[1].as_str() == "M" && score < m_score {
            m_score = score;
            m_id = v[2].clone();
            m = v[0].clone();
        }
        if v[1].as_str() == "F" && score > f_score {
            f_score = score;
            f_id = v[2].clone();
            f = v[0].clone();
        }
        input.clear();
    }
    if f.is_empty() {
        println!("Absent");
    } else {
        println!("{} {}", f, f_id);
    } 
    if m.is_empty() {
        println!("Absent");
    } else {
        println!("{} {}", m, m_id);
    }

    if !m.is_empty() && !f.is_empty() {
        println!("{}", f_score - m_score);
    } else {
        println!("NA");
    }
    //println!("{} {}", m, f);
}
